### DISCUSSION TITLE
sensor timestamp in Libre View

### DISCUSSION THREAD

**Discussion by source2k** [j-kaltes/Juggluco#56] [Ideas]
I used the original Abbot app with German Libre 3 sensors until a few days.
I read the current BG value and a sensor timestamp from https://api-de.libreview.io/llu/connections.
After adding 14 days to the timestamp, I get the sensor end date.

The timestamp value is stored in "a" of the "sensor" variable in the JSON response:
"sensor":
{
 "deviceId":"",
 "sn":"0XXXXXXXX0",
 "a":9999999999,
 "w":60,
 "pt":4,
 "s":false,
 "lj":false
}

I switched to Juggluco (great software) and the "a" timestamp was updated to the time, when I re-scanned the sensor to take it over to Juggluco.

The question is, what should the variable contain. The last scan time or the initial sensor start time.
I'm asking because I would prefer the initial start time :)

**Comment by j-kaltes** [comment]
> I used the original Abbot app with German Libre 3 sensors until a few days. I read the current BG value and a sensor timestamp from https://api-de.libreview.io/llu/connections. After adding 14 days to the timestamp, I get the sensor end date.
> 
> The timestamp value is stored in "a" of the "sensor" variable in the JSON response: "sensor": { "deviceId":"", "sn":"0XXXXXXXX0", "a":9999999999, "w":60, "pt":4, "s":false, "lj":false }
> 
> I switched to Juggluco (great software) and the "a" timestamp was updated to the time, when I re-scanned the sensor to take it over to Juggluco.
> 
> The question is, what should the variable contain. The last scan time or the initial sensor start time. I'm asking because I would prefer the initial start time :)

Juggluco sends the sensor start time to Libreview, that is all what Juggluco does. Where the data structure you are talking about comes from, I don't know..

**Reply by source2k** [reply]
The data structure comes from Abbot's LibreView server.
The server can be queried in the same way like done by Abbot's LibreLinkUp follower app.
The advantage is, that the server is always available.

I though that the time stamp is set, when the initiating app is sending the start information.
Then it would have been possible, that this is the date of first sensor start instead of the last scan date.
